tests/backup/backup_psa_test.go

```diff
@@ -1,30 +1,551 @@
+			labelSelectors := make(map[string]string)
+			for backupLocationUID, backupLocationName := range backupLocationMap {
+				for _, namespace := range bkpNamespaces {
+					wg.Add(1)
+					go func(namespace string) {
+						defer wg.Done()
+						backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+						appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+						err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, backupLocationName, backupLocationUID, appContextsToBackup, labelSelectors, BackupOrgID, sourceClusterUid, "", "", "", "")
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupName))
+						backupNames = append(backupNames, backupName)
+					}(namespace)
+				}
+			}
+			wg.Wait()
+		})
+
+		Step("Create new storage class for restore", func() {
+			err := SetDestinationKubeConfig()
+			dash.VerifyFatal(err, nil, "Setting destination kubeconfig")
+
+			log.InfoD("Create new storage class for restore")
+			scName = fmt.Sprintf("replica-sc-%v", time.Now().Unix())
+			params["repl"] = "1"
+			k8sStorage := storage.Instance()
+			v1obj := metaV1.ObjectMeta{
+				Name: scName,
+			}
+			reclaimPolicyDelete := v1.PersistentVolumeReclaimDelete
+			bindMode := storageApi.VolumeBindingImmediate
+			scObj := storageApi.StorageClass{
+				ObjectMeta:        v1obj,
+				Provisioner:       k8s.CsiProvisioner,
+				Parameters:        params,
+				ReclaimPolicy:     &reclaimPolicyDelete,
+				VolumeBindingMode: &bindMode,
+			}
+			_, err = k8sStorage.CreateStorageClass(&scObj)
+			dash.VerifyFatal(err, nil, "Verifying creation of new storage class")
+
+			err = SetSourceKubeConfig()
+			dash.VerifyFatal(err, nil, "Setting source kubeconfig")
+		})
+
+		Step("Default restore of applications by replacing the existing resources with NS level PSA", func() {
+			log.InfoD("Default restore of applications by replacing the existing resources with NS level PSA")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			var wg sync.WaitGroup
+			for _, backupName := range backupNames {
+				wg.Add(1)
+				go func(backupName string) {
+					defer wg.Done()
+					defaultRestoreName := fmt.Sprintf("%s-%s-default", RestoreNamePrefix, backupName)
+					err = CreateRestoreWithReplacePolicyWithValidation(defaultRestoreName, backupName, make(map[string]string), SourceClusterName, BackupOrgID, ctx, make(map[string]string), ReplacePolicyDelete, scheduledAppContexts)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creating default restore for backup with replace policy [%s]", defaultRestoreName))
+					restoreNames = append(restoreNames, defaultRestoreName)
+				}(backupName)
+			}
+			wg.Wait()
+		})
+
+		Step("Restore of applications with NS and StorageClass mapping with NS level PSA", func() {
+			log.InfoD("Restore of applications with NS and StorageClass mapping with NS level PSA")
```

Comment:
In this case all the restored ns will have no PSA labels
Can we create restored ns with PSA label before restoring and the use the ns for ns mapping?

Reply:
Can I add a separate step for it?

Reply:
Done